### DISCUSSION TITLE
Usernames are Evolving - Revolt

### DISCUSSION THREAD

**Discussion by giscus** [stoatchat/discussions#554] [Announcements]
# Usernames are Evolving - Revolt

Now more personal than ever.

https://revolt.chat/posts/evolving-usernames

**Comment by SnenxyTengoku** [comment]
# :gigachad:

**Comment by IryRasmo** [comment]
GOOD SHIT BOYS

**Comment by enderman0125** [comment]
this is fucking hilarious to me
based fucking revolt

**Comment by cngld** [comment]
I love how you used the same "usernames are evolving" as Discord did. +1 to you guys. Porting my community server from Discord to Revolt.

**Comment by veselle** [comment]
Hello, is this the based department?
...Oh, the call is for you guys.

**Comment by Elsie19** [comment]
👏

**Comment by ghost** [comment]
O exato oposto do Discord, não tinha como ser melhor do que isso

**Comment by thehatkid** [comment]
### Revolt W

**Comment by LightCyan01** [comment]
revolt w

**Comment by AriaLyndis** [comment]
LOVE This! Will we be able to choose our own discriminator's if available? My old number from discord, though arbitrary, holds a lot of sentimental value. I'd be eager to pay for my old number

**Reply by ImaTimekeeperReadingRaptor** [reply]
I'd love this, I'd pay since Revolt seems like they'd listen to the community!

**Reply by recatek** [reply]
Agreed, I like having a specific discriminator with my name if it's available. I'd like to be able to customize mine here.

**Reply by SnenxyTengoku** [reply]
i think its already been decided that you won't be able to, because being able to change it defeats one of the main points of having discriminators, the only way it can change is by setting a username that may conflict with another username

**Reply by MasicoreLord** [reply]
They could allow users to set their own discrim, and avoid conflicts at the same time.

**Reply by sussycatgirl** [reply]
We don't intend to allow users to set a custom discriminator - The idea is to have them be completely random so it's fair for everyone. Not even Revolt staff/developers get to choose, unlike on Discord :)

**Reply by Xalphire** [reply]
> We don't intend to allow users to set a custom discriminator - The idea is to have them be completely random so it's fair for everyone. Not even Revolt staff/developers get to choose, unlike on Discord :)

Yeah that makes sense! Y'all actually have a brain unlike Discord

**Comment by ghost** [comment]
Discriminators on revolt, good job team :)

**Comment by BrydenIsNotSmart** [comment]
This was a really good move by the developers of Revolt, hundreds if not thousands of people joined Revolt just today.

**Comment by ImaTimekeeperReadingRaptor** [comment]
Love when Discord alternative gains a feature Discord ripped away

**Comment by copmidsize** [comment]
это так называемая база

**Comment by searinminecraft** [comment]
**Discord**: Removes discriminators
**Revolt**: We are adding discriminators ~~so people won't call Revolt a Discord clone~~

(still a banger update though)

**Comment by Timtonix** [comment]
I need to bring my fellow to Revolt x)

**Comment by DeclanChidlow** [comment]
This is bloody sick. Thanks as always Revolt team!

**Comment by YSCCL** [comment]
Oh yeah, this solidifies it. This will be the place I will tell my fellow buds to migrate to

**Comment by ert994** [comment]
this is fucking awesome.

**Comment by ijsbol** [comment]
bros a legend for doing this 💀

**Comment by aeongdesu** [comment]
discord 🔄 revolt
they're trading with username system 🔥🔥🔥🔥

**Comment by arslee07** [comment]
^

**Comment by NatoBoram** [comment]
About the new native apps; so you foresee a situation where both apps might have different sets of features due to the difference between the people that code in Swift vs Kotlin? 

Historically, all apps I've used that have two native apps have had this problem in one way or another.

**Reply by cyrneko** [reply]
Whilst I would love platform native clients, I would assume they might use Flutter, React Native, or a similar tool.

I'd love to be proven wrong though!

I also don't think they would massively delay features between platforms.

**Reply by Rexogamer** [reply]
We're currently working on fully native apps (@infi os working on a Kotlin Android app; @zomatree is working on a Swift iOS app)

**Comment by MPS64** [comment]
this app is going to be peak!!!!!!!!

**Comment by Andre601** [comment]
Would be good if the RFC was linked in the post for those who want to read the nerdy details. 🤓

**Reply by DeclanChidlow** [reply]
Here it is!
https://github.com/revoltchat/rfcs/pull/3

**Comment by Jonathing** [comment]
unfathomably based

**Comment by Nightcaat** [comment]
The absolute shade. You love to see it

**Comment by Rainstar760** [comment]
ABSOLUTELY BASED

**Comment by rainyskye** [comment]
Holy shit, I might actually check out Revolt looool

**Comment by ghost** [comment]
Yet another epic W for Revolt.

**Comment by VBlaks** [comment]
Why are there now display names and usernames with a discriminator??? Isn't the whole point of the discriminator so you can pick any username and don't need the displayname

**Reply by Andre601** [reply]
The point was explained in the post.
The discrim allows multiple users with same name, while display name allows more special characters.
(Username only allows alphanummeric unicode text iirc)

**Reply by VBlaks** [reply]
It would be cool if users with the same display name as their username just didn't have a username, my username and displayname are the same but it still shows me both of them

**Comment by Xalphire** [comment]
Wait but how do I change the discriminator? It's just greyed out for me.

**Reply by DeclanChidlow** [reply]
There is no way to change discriminators.

**Reply by Xalphire** [reply]
As of now or forever? I think I might be the only "Xalphire" out there loll

**Reply by DeclanChidlow** [reply]
Maybe not forever but there is nothing planned at the moment, at least not to my knowledge.

**Reply by cyrneko** [reply]
> As of now or forever? I think I might be the only "Xalphire" out there loll

Probably forever, as discriminators aren't supposed to be for style.

They're supposed to be fixed and 100% utility.

**Comment by Xalphire** [comment]
Anyways Discord -1 Revolt +1

**Comment by ghost** [comment]
This is revolting.

**Comment by vokap** [comment]
amazing

**Comment by sophuric** [comment]
gigachad revolt

**Comment by Sienna7002** [comment]
i love how they are going from discord's new naming system to the arguably much better older one
# absolute W

**Comment by davidvkimball** [comment]
What a huge mistake. I'm sure they regret this decision now.

**Reply by Xalphire** [reply]
What why

**Reply by davidvkimball** [reply]
No one has been able to explain to me how forced discriminators (especially ones you can't change) is somehow a better system than traditional usernames. Discord's username fix was pretty much the only good change they've made in the last 4 years. This is something Revolt Chat had in the beginning, and then they changed for what? A flash-in-the-pan PR stunt?

**Reply by Xalphire** [reply]
By the way I completely forgot about the existence of Revolt until you replied lol

**Reply by davidvkimball** [reply]
Lol. Someone replied to my tweet about this change, and it reminded me about it. So happy to pass along the ripple effect!

**Reply by Andre601** [reply]
Discriminator allow multiple people to have the same username.

Like imagine you want the username "David" on Discord or Twitter. Chances are extremely high that this name is already taken, so you would either need to choose a completely different name, or something like "David1", "David2", ...

With Discriminators, you can just get "David" and the 4 digit number makes sure it remains a unique username.

It just allows more people to have the username of their choice.

**Reply by davidvkimball** [reply]
Hey @Andre601 - appreciate the reply. I've heard that sentiment before, but it makes no sense to me. 

With discriminators, no one gets "David" as a username. The best you can get is something like "David#0001" which is arguably worse than a simple David1. 

A discriminator system also punishes people who curated a unique username and brand for themselves. 

"Hey, I want to add you on Revolt Chat." 
"Sure, my username is David!" 
"Oh, that's strange. There are a million results, and your picture doesn't come up for me. Can you double check your name?"
"Oops, I'm actually David#2854." 
"Oh..." 

So again, literally no one gets the name they want, instead of at least one person getting the name they want - and on top of that, with  the discriminator system, literally everyone is stuck with a 4-number suffix (that in some cases, you can't change). Objectively worse all-around.

**Reply by Andre601** [reply]
They do get the name. The Username#1234 is the tag, consisting of the username and the discriminator.
So technically speaking do you get your username here.

When discord removed discrimators was it the worst decision ever... Now when revolt adds it is it the worst decision ever (for you).
You people seriously can't be satisfied.

**Reply by insertish** [reply]
I would like to just mention that we were always planning to implement discriminators, [we've been having this discussion since 2021](https://github.com/orgs/revoltchat/discussions/61) (and it ended up as one of the more favoured feature requests) and [then again we involved the community when making this change](https://github.com/revoltchat/rfcs/pull/3), Discord's decision to drop them was just more reason to make the change when we did. I still think we made the right call.

**Reply by DeclanChidlow** [reply]
I think it's important to note that the majority of users appreciated this change, and a lot of people joined the platform for it. It's unfortunate the love isn't unanimous, but, as Insert said, I think the right call was made.

**Reply by davidvkimball** [reply]
Hey @insertish, thanks for the reply. It certainly had a vocal group speaking out against it, interesting to see it's something users really want. 

@Andre601 I, apparently, am in the minority for never wanting discriminators, ever. I was as loud and opposed to it when I first made my Discord account in June 2015 as I am now. I find it interesting how, even for you, if you wanted consistency, the best you could do is Andre#0601, since Anddre601 is your GitHub username. Still one character longer than what you have.

**Reply by Andre601** [reply]
Funfact: On Discord did I have `Andre_601#0601` when they had discriminators and Nitro allowed you to have custom ones...

For me, the number doesn't matter, as the username is all that counts for me. If I can be `Andre_601`, I'm okay with it. It's better than being something like `RealAndre_601` like on Twitter because some old not used account to my favourited name first...

The discriminator btw also eliminates name stealing/selling. On Discord, names still get sold for stupid amounts due to their name, length, etc... It was and still is a major issue there and on every platform utilizing the pomelo format. Using a discrim, a name no longer is no longer 100% unique, so there is less insentive of sniping a name to then sell.

**Reply by davidvkimball** [reply]
@Andre601 that is very good to know - thank you for sharing that. 

I'm not sure what your professional background is, but I'm a digital marketer, and branding / consistency is extremely important to me. 

So with all of the brands I create (including personal brands) I am very thoughtful about picking a unique brand name for unique username purposes. 

I would also say that, while your point about discriminators reducing stealing/selling might be true, that is not a good reason to suggest discriminators are good - it just means usernames aren't as important. I'd also argue that the same level of selling / trading still happened with the [username]#0001 names, which effectively became the "de-facto default username option" from a branding standpoint. In general, you shouldn't judge a thing based on its abuse.

**Reply by IAmTomahawkx** [reply]
I'd have to disagree with you on the abuse. Because of how discords system worked, the tag (username#discrim) was quite literally never part of a sale, because it could be changed to anything so easily. You could pick any name you wanted, and if you wanted extra clout you'd pop nitro on the account to get a special discriminator (or buy an account that had a year of nitro applied or whatever). Also there were a lot of different "clout" discriminators, I commonly saw anything from 0001 to 1000 to 1111, and so on. Notably, these were all done via nitro, if the nitro went bye bye so did your fancy discriminator.
Nowadays usernames are a feature point on discord account sales because you cannot get any name you want. 

As a platform, abuse is an *extremely* important and valid reason to implement changes to our systems. Having your account stolen because you have a unique name is an [incredibly large issue](https://gimletmedia.com/amp/shows/reply-all/v4he6k) on other platforms, and reducing the risk of targeted attacks (especially to people who aren't public figures, such as the person in the podcast episode linked above) is essential. This may come at the (in our eyes as a platform) small cost of personal branding, however you can easily work around this on your servers by creating roles that indicate who's "the real account". 
And if you're worried about fake accounts under your brand name, this happens irrelevant to what naming system is in use. Discord servers I moderate consistently receive impersonation accounts of larger streamers and YouTubers that simply change letters to similar looking numbers, or just adding in letters (stearmsupport, anyone?).

Side note but the bigger thing people were and are still after are badges.

tldr: Abuse is an extremely important angle to consider for us as a platform, and it trumps the branding use case.
